pkg/tsdbctl/check.go

```diff
@@ -91,58 +126,172 @@ func (cc *checkCommandeer) check() error {
 	}
 
 	// Get metric data and metadata
-	allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
 	container, tablePath := cc.rootCommandeer.adapter.GetContainer()
-	objPath := fmt.Sprintf("%s/%s", tablePath, cc.objPath)
+
+	var respChans []chan *v3io.Response
+	if cc.metricName == "" {
+		respChan, err := cc.checkByPath(container, tablePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check using object path '%s'", cc.objPath)
+		}
+		respChans = []chan *v3io.Response{respChan}
+		objPath := "/" + path.Join("", tablePath, cc.objPath)
+		allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
+		input := v3io.GetItemInput{Path: objPath, AttributeNames: allAttrs}
+		_, err = container.GetItem(&input, nil, respChan)
+		if err != nil {
+			return errors.Wrap(err, "failed to send GetItem request")
+		}
+	} else {
+		respChans, err = cc.checkByName(container, tablePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check using metric name (%s) and labels (%v)", cc.metricName, cc.labels)
+		}
+	}
+
+	for _, respChan := range respChans {
+		resp := <-respChan
+		if resp.Error != nil {
+			return errors.Wrap(resp.Error, "GetItem returned with an error")
+		}
+		err = cc.printResponse(resp)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cc *checkCommandeer) checkByPath(container *v3io.Container, tablePath string) (chan *v3io.Response, error) {
+	var err error
+	respChan := make(chan *v3io.Response, 1)
+	objPath := "/" + path.Join("", tablePath, cc.objPath)
+	allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
 	input := v3io.GetItemInput{Path: objPath, AttributeNames: allAttrs}
-	resp, err := container.Sync.GetItem(&input)
+	_, err = container.GetItem(&input, nil, respChan)
 	if err != nil {
-		return errors.Wrap(err, "GetItem failed.")
+		return nil, errors.Wrap(err, "failed to send GetItem request")
+	}
+	return respChan, nil
+}
+
+func (cc *checkCommandeer) checkByName(container *v3io.Container, tablePath string) ([]chan *v3io.Response, error) {
+
+	var err error
+
+	schema, err := getSchema(cc.rootCommandeer.v3iocfg, container)
+	if err != nil {
+		return nil, err
+	}
+	numBuckets := schema.TableSchemaInfo.ShardingBucketsCount
+	hash := cc.labels.Hash()
+	var from, to int64
+	if cc.from == "" {
+		from = 0
+	} else {
+		from, err = utils.Str2unixTime(cc.from)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse start time")
+		}
+	}
+	if cc.to == "" {
+		to = math.MaxInt64
+	} else {
+		to, err = utils.Str2unixTime(cc.to)
+		return nil, errors.Wrap(err, "failed to parse end time")
+	}
+	partitionInterval, _ := utils.Str2duration(schema.PartitionSchemaInfo.PartitionerInterval)
+	var respChans []chan *v3io.Response
+	for _, partition := range schema.Partitions {
+		partitionEndTime := partition.StartTime + partitionInterval
+		if partitionEndTime >= from && partition.StartTime <= to {
```

Comment:
the condition looks wrong to me. 
IMHO it should be `partition.StartTime >= from &&  partitionEndTime <= to`

Reply:
When `from` or `to` falls in the middle of a partition, we want to include that partition.